packages/react-native-gesture-handler/apple/RNGestureHandlerDetector.mm

```diff
@@ -0,0 +1,147 @@
+    _props = defaultProps;
+    _moduleId = -1;
+  }
+
+  return self;
+}
+
+// TODO: I'm not sure whether this is the correct place for cleanup
+// Possibly allowing recycling and doing this in prepareForRecycle would be better
+- (void)willMoveToWindow:(UIWindow *)newWindow
+{
+  if (newWindow == nil) {
+    RNGestureHandlerManager *handlerManager = [RNGestureHandlerModule handlerManagerForModuleId:_moduleId];
+    react_native_assert(handlerManager != nullptr && "Tried to access a non-existent handler manager")
+        const auto &props = *std::static_pointer_cast<const RNGestureHandlerDetectorProps>(_props);
+
+    for (const auto handler : props.handlerTags) {
+      NSNumber *handlerTag = [NSNumber numberWithInt:handler];
+      [handlerManager.registry detachHandlerWithTag:handlerTag];
+    }
+  }
+}
+
+- (void)dispatchStateChangeEvent:(RNGestureHandlerDetectorEventEmitter::OnGestureHandlerStateChange)event
+{
+  if (_eventEmitter != nullptr) {
+    std::dynamic_pointer_cast<const RNGestureHandlerDetectorEventEmitter>(_eventEmitter)
+        ->onGestureHandlerStateChange(event);
+  }
+}
+
+- (void)dispatchGestureEvent:(RNGestureHandlerDetectorEventEmitter::OnGestureHandlerEvent)event
+{
+  if (_eventEmitter != nullptr) {
+    std::dynamic_pointer_cast<const RNGestureHandlerDetectorEventEmitter>(_eventEmitter)->onGestureHandlerEvent(event);
+  }
+}
+
+- (void)dispatchTouchEvent:(RNGestureHandlerDetectorEventEmitter::OnGestureHandlerTouchEvent)event
+{
+  if (_eventEmitter != nullptr) {
+    std::dynamic_pointer_cast<const RNGestureHandlerDetectorEventEmitter>(_eventEmitter)
+        ->onGestureHandlerTouchEvent(event);
+  }
+}
+
+#pragma mark - RCTComponentViewProtocol
+
++ (ComponentDescriptorProvider)componentDescriptorProvider
+{
+  return concreteComponentDescriptorProvider<RNGestureHandlerDetectorComponentDescriptor>();
+}
+
+- (void)updateLayoutMetrics:(const LayoutMetrics &)layoutMetrics
+           oldLayoutMetrics:(const LayoutMetrics &)oldLayoutMetrics
+{
+  auto newLayoutMetrics = layoutMetrics;
+  // Override to force hittesting to work outside bounds
+  newLayoutMetrics.overflowInset = {.left = 1, .right = 1, .top = 1, .bottom = 1};
+
+  [super updateLayoutMetrics:newLayoutMetrics oldLayoutMetrics:oldLayoutMetrics];
+}
+
+- (void)updateProps:(const Props::Shared &)propsBase oldProps:(const Props::Shared &)oldPropsBase
+{
+  const auto &newProps = *std::static_pointer_cast<const RNGestureHandlerDetectorProps>(propsBase);
+  const auto &oldProps = *std::static_pointer_cast<const RNGestureHandlerDetectorProps>(oldPropsBase);
+
+  _moduleId = newProps.moduleId;
+  RNGestureHandlerManager *handlerManager = [RNGestureHandlerModule handlerManagerForModuleId:_moduleId];
+  react_native_assert(handlerManager != nullptr && "Tried to access a non-existent handler manager")
+
+      std::unordered_map<int, RNGestureHandlerMutation>
+          changes;
```

Comment:
Looks weird, have you run `yarn format:apple`?

Reply:
Yes